pages/form.tsx

```diff
@@ -17,93 +19,117 @@ const preventMinus = (e: React.KeyboardEvent<HTMLInputElement>) => {
-            solarPanelCapacity: parseFloat(
-              formData.get('solarPanelCapacity') as string
-            ),
-          };
-        } catch (e) {
-          console.error(`onSubmit(): ${e}`);
-        }
-      }}
-    >
-      <div>
-        <h1 className="font-bold text-4xl mt-20 dark:text-ocf-gray">
-          Panel Details
-        </h1>
-      </div>
-      <div className="my-8">
-        <h2 className={h2ClassName}>Solar panel direction</h2>
-        <p className={pClassName}>
-          (0º = North, 90º = East, 180º = South, 270º = West)
-        </p>
-        <input
-          required
-          name="solarPanelDirection"
-          type="number"
-          placeholder="170º"
-          onKeyDown={preventMinus}
-          min="0"
-          max="360"
-          className={`${inputClassName} w-20`}
-        />
-        <p className={`${pClassName} underline`}>I don&apos;t know</p>
-      </div>
+      <Input
+        id="solar-panel-direction"
+        label="Solar panel direction"
+        description="(0º = North, 90º = East, 180º = South, 270º = West)"
+        help="I don't know"
+        inputProps={{
+          type: 'number',
+          placeholder: '170º',
+          min: '0',
+          max: '360',
+          required: true,
+          onKeyDown: preventMinus,
+        }}
+      />
 
-      <div className="my-8">
-        <h2 className={h2ClassName}>Solar panel angle tilt</h2>
-        <p className={pClassName}>(Degrees above the horizontal)</p>
-        <input
-          required
-          name="solarPanelAngleTilt"
-          type="number"
-          placeholder="30º"
-          min="0"
-          max="360"
-          onKeyDown={preventMinus}
-          className={`${inputClassName} w-20`}
-        />
-      </div>
+      <Input
+        id="solar-panel-tilt"
+        label="Solar panel tilt"
+        description="(Degrees above the horizontal)"
+        inputProps={{
+          type: 'number',
+          placeholder: '30º',
+          min: '0',
+          max: '360',
+          required: true,
+          onKeyDown: preventMinus,
+        }}
+      />
 
-      <div className="my-8">
-        <div className="flex flex-row justify-start items-center w-full">
-          <h2 className={h2ClassName}>Solar panel capacity</h2>
-          <p className={pClassName}>(optional)</p>
-        </div>
-        <input
-          name="solarPanelCapacity"
-          type="number"
-          placeholder="3000 W"
-          min="0"
-          onKeyDown={preventMinus}
-          className={`${inputClassName} w-28`}
-        />
-      </div>
+      <Input
+        label={
+          <>
+            Solar panel capacity
+            <span className="text-xs font-normal">(optional)</span>
+          </>
+        }
+        id="solar-panel-capacity"
+        inputProps={{
+          type: 'number',
+          placeholder: '30º',
+          min: '0',
+          max: '360',
+          onKeyDown: preventMinus,
+        }}
+      />
 
```

Comment:
currently, none of these inputs allow floats. we might need to think about if this should be expected behavior or not, but if we want to allow floats, I think what we need to do is pass in `step: 'any'` to our `<input>` elements. since `step=1` as a default if not specified, I think our current input fields are throwing `stepMismatch` violations with non-integer inputs.